File: Santiq/Models/Alert.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Santiq.Models
{
    public class Alert
    {
        public int Id { get; set; }
        public bool IsThreat { get; set; }
        public string UserFirstName { get; set; }
        public string UserLastName { get; set; }
        public string ServiceKey { get; set; }
        public string UserId { get; set; }
        public bool UserIsChild { get; set; }
        public string GeneratedOn { get; set; }
        public string AcknowledgedOn { get; set; }
        public int Viewed { get; set; }
        public string AlertTypeName { get; set; }
        public string Category { get; set; }
        public string AvailableSubResource { get; set; }
        public string Status { get; set; }
        public string Description { get; set; }
        public string Title { get; set; }
        public string DisplayName { get; set; }
        public string ResolveDate { get; set; }
        public string ThreatDate { get; set; }
        public string ServiceType { get; set; }
        public string CreatedDate { get; set; }
        public string CreatedTime { get; set; }
        public string Type { get; set; }
        public string FirstViewDate { get; set; }
        public bool HasError { get; set; }
        public string CaseId { get; set; }
        public string AlertCategory { get; set; }
        public string[] Errors { get; set; }
        public string[] TitleFields { get; set; }
        public string[] MonitorTypes { get; set; }
    }
}

File: Santiq/ClientApp/src/store/Alert.ts
import { Action, Reducer } from 'redux';
import { AppThunkAction } from './';

// -----------------
// STATE - This defines the type of data maintained in the Redux store.

export interface AlertState {
    isLoading: boolean;
    startDateIndex?: number;
    alerts: Alert[];
}

export interface Alert {
    id: number;
    isThreat: boolean;
    userFirstName: string;
    userLastName: string;
    serviceKey: string;
    userId: string;
    userIsChild: boolean;
    generatedOn: string;
    acknowledgedOn: string;
    viewed: number;
    alertTypeName: string;
    category: string;
    availableSubResource: string;
    status: string;
    description: string;
    title: string;
    displayName: string;
    resolveDate: string;
    threatDate: string;
    serviceType: string;
    createdDate: string;
    createdTime: string;
    type: string;
    firstViewDate: string;
    hasError: boolean;
    caseId: string;
    alertCategory: string;
    errors: any;
    titleFields: any;
    monitorTypes: any;
}

interface RequestAlertsAction {
    type: 'REQUEST_ALERTS';
    startDateIndex: number;
}

interface ReceiveAlertsAction {
    type: 'RECEIVE_ALERTS';
    startDateIndex: number;
    alerts: Alert[];
}


type KnownAction = RequestAlertsAction | ReceiveAlertsAction;

export const actionCreators = {
    requestAlerts: (startDateIndex: number): AppThunkAction<KnownAction> => (dispatch, getState) => {

        


        const appState = getState();

        console.log(startDateIndex);
        console.log(appState);


        if (appState && appState.alerts && startDateIndex !== appState.alerts.startDateIndex) {
            fetch(`api/alerts?pageNo=${startDateIndex}`)
                .then(response => response.json() as Promise<Alert[]>)
                .then(data => {
                    dispatch({ type: 'RECEIVE_ALERTS', startDateIndex: startDateIndex, alerts: data });
                });

            dispatch({ type: 'REQUEST_ALERTS', startDateIndex: startDateIndex });
        }
    }
};

const unloadedState: AlertState = { alerts: [], isLoading: false };

export const reducer: Reducer<AlertState> = (state: AlertState | undefined, incomingAction: Action): AlertState => {
    if (state === undefined) {
        return unloadedState;
    }

    const action = incomingAction as KnownAction;
    switch (action.type) {
        case 'REQUEST_ALERTS':
            return {
                startDateIndex: action.startDateIndex,
                alerts: state.alerts,
                isLoading: true
            };
        case 'RECEIVE_ALERTS':
            return {
                startDateIndex: action.startDateIndex,
                alerts: action.alerts,
                isLoading: false
            };
        default:
            break;
    }

    return state;
};
